File: 프로그래머스/lv2/12981. 영어 끝말잇기/영어 끝말잇기.py
import math
def solution(n, words):
    answer = [0, 0]
    length = len(words)
    prev_word = [words[0]]
    
    for i in range(1, length):
        order = (i % n) + 1
        game = (i // n) + 1
        print(order, game)
        
        if prev_word[-1][-1] != words[i][0] or words[i] in prev_word:
            answer = [order, game]
            break
        else:
            prev_word.append(words[i])
    
    return answer

File: 프로그래머스/2/12981. 영어 끝말잇기/영어 끝말잇기.js
function solution(n, words) {
    let answer = [0, 0];
    const history = new Set()
    for (let i = 0; i < words.length; i++) {
        const word = words[i]
        if (i === 0) {
            history.add(word)
            continue
        }
        
        const prev = words[i - 1]
        const player = i % n + 1
        const round = Math.floor(i / n) + 1
        
        if (history.has(word) || prev.at(-1) !== word[0]) return [player, round]
        history.add(word)
    }

    return answer;
}